src/typing/typeload.ml

```diff
@@ -318,14 +318,18 @@ let rec load_params ctx info params p =
 			let c = mk_class ctx.m.curmod ([],name) p (pos e) in
 			c.cl_kind <- KExpr e;
 			TInst (c,[]),pos e
-		| TPType t ->
+		| TPType (CTPath({ path = { tpackage = ["$"]; tname = "_hx_default" }}),p), None ->
+			raise_typing_error "Cannot apply default type parameter on non-default type parameter" p
+		| TPType (CTPath({ path = { tpackage = ["$"]; tname = "_hx_default" }}),p), Some def ->
+			def,p
+		| TPType t, _ ->
```

Comment:
That's an awkward match, would be better to keep the `match t` and then have an inner `match def with` for the two cases here.

Reply:
Indeed :+1: